libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteArrow.java

```diff
@@ -108,6 +108,9 @@
   }
 
   void addUpcomingManeuverArrow(RouteProgress routeProgress) {
+    if (routeProgress.route() == null) {
```

Comment:
is it possible that the current MapRouteArrow is visible, and then it won't be changed to invisible? Wondering if we can have a stale MapRouteArrow on the map

https://github.com/mapbox/mapbox-navigation-android/pull/2755/files#diff-259e8a0b74d93ac6d8040ce735fc2e7fR117-R122

Reply:
I don't think so. Also as a new feature request we are going to clear the entire part of the route that has been traversed hence the route arrow that already has been traversed will go away as well.

Reply:
then it may be worth moving this fix to `MapRouteProgressChangeListener` where it's set up to be tested.

otherwise can you add tests to `MapRouteArrow`?